src/software/new_geom/polynomial1d.cpp

```diff
@@ -53,7 +53,7 @@ unsigned int Polynomial1d::getOrder() const
 {
     if (coeffs.size() != 0)
     {
-        for (size_t i = coeffs.size() - 1; i >= 0; i--)
+        for (int i = coeffs.size() - 1; i >= 0; i--)
```

Comment:
why not just iterate in the other direction? I suppose it's likely that we're going to need to iterate over a few more elements, but I think the overall performance hit seems pretty minimal?

Reply:
Another option is to offset `i` by 1 so you can iterate until `i > 0` instead. 

Also we noticed during in-person discussion that we're doing a signed - unsigned comparison here.

Reply:
I did the offset i by 1 thing. it's way more clean